### ISSUE TITLE
Start indexing at 0

### ISSUE THREAD

**Issue by dpsanders** [JuliaDiff/TaylorSeries.jl#133]
The `Polynomials.jl` package starts indexing at `0`, so that `p[0]` is the 0th order term, `p[1]` is the first-order term, etc. (see below).

I suggest changing to this behaviour.

```jl
julia> using Polynomials

julia> p = Poly([1, 2, 3])
Poly(1 + 2*x + 3*x^2)

julia> p[0]
1

julia> p[1]
2

julia> p[2]
3

julia> p[3]
0
```

**Comment by lbenet** [comment]
I think this is easy to implement (a simple shift in the definition), and may be it is good to stick to the same convention. But it is a breaking change, thinking simply on `TaylorIntegration.jl`.

One comment: currently, the equivalent of your example with `TaylorSeries.jl` (putting aside the issue on the 0-index or 1-index) yields the following:
```julia
julia> using TaylorSeries

julia> q = Taylor1([1,2,3])
 1 + 2 t + 3 t² + 𝒪(t³)

julia> q[1]
1

julia> q[4]
ERROR: BoundsError: attempt to access 3-element Array{Int64,1} at index [4]
Stacktrace:
 [1] getindex(::TaylorSeries.Taylor1{Int64}, ::Int64) at /Users/benet/.julia/v0.6/TaylorSeries/src/auxiliary.jl:77
```
The point I want to stress is the fact that `q[4]` throws an error, instead of being zero. I think we discussed this somewhere, and it was decided that way, but i don't recall the discussion.

Cc: @PerezHz

**Comment by lbenet** [comment]
While it is trivially a shift, it matters a lot internally!

**Comment by PerezHz** [comment]
I agree that this change would conform even more TaylorSeries syntax to standard mathematical notation, so I'd say let's go for it! Perhaps even more importantly, this change allows the user (or at least me) to "save" one mental step when dealing with TaylorSeries polynomials. So, even if that means changing a lot of internal stuff both in TaylorSeries and TaylorIntegration, I think it's worthwhile to implement this!

**Comment by lbenet** [comment]
Solved by #135. Closing
